### PR TITLE
SSTOOLKIT-101

### PR DESCRIPTION
**Pull request by bertofl90**

### CODE REVIEW

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ Doc. ID: XRDSST-CONF
 | 06.07.2021 | 2.1.1       | Add client unregister command                                                | Alberto Fernandez  |
 | 06.07.2021 | 2.1.2       | Add client delete command                                                    | Alberto Fernandez  |
 | 09.07.2021 | 2.1.3       | Add listing and deletion of access rights for services                       | Bert Viikmäe       |
+| 06.07.2021 | 2.1.4       | Add local groups management                                                  | Alberto Fernandez  |
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md]: Date should reflect today

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 2]
```diff
@@ -365,6 +374,11 @@ security_server:
       connection_type: <CONNECTION_TYPE>
       tls_certificates:
   		- /path/to/tls_cert
+      local_groups:
+        - code: <LOCAL_GROUP_CODE>
+          description: <LOCAL_GROUP_DESCRIPTION>
+          members:
+            - <SERVICE_CLIENT_ID>
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md]: In the configuration file you have "<MEMBER_ID>"

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 3]
```diff
@@ -1119,6 +1143,91 @@ xrdsst member list-classes --instance <XROAD-INSTANCE>
...
+##### 4.2.10.3 Local groups list
+
+* Access rights: XROAD_SYSTEM_ADMINISTRATOR
+
+List client local groups can be done with:
+```
+xrdsst local-group list --ss <SECURITY_SERVER_NAME> --client <CLIENT_ID>
+```
+* <SECURITY_SERVER_NAME> seccurity server name, e.g. ss1
+* <CLIENT_ID> subsystem client id, e.g. DEV:COM:12345:COMPANY
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md:1183]: Do you also support multiple clients or just a single one?

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 4]
```diff
@@ -1119,6 +1143,91 @@ xrdsst member list-classes --instance <XROAD-INSTANCE>
...
+╒══════╤════════════╤════════════════════════╤═════════════════════════════════════════════════════════╕
+│   ID │ CODE       │ DESCRIPTION            │ MEMBERS                                                 │
+╞══════╪════════════╪════════════════════════╪═════════════════════════════════════════════════════════╡
+│  185 │ OtherGroup │ description 1          │ []                                                      │
+├──────┼────────────┼────────────────────────┼─────────────────────────────────────────────────────────┤
+│  125 │ TestGroup  │ Description test group │ ['DEV:GOV:9999:SUBGOV', 'DEV:ORG:0000:SUBORGANIZATION'] │
+╘══════╧════════════╧════════════════════════╧═════════════════════════════════════════════════════════╛
+
+
+The table above shows the following information about the certificates:
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md]: "certificates" is wrong here

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 5]
```diff
@@ -1119,6 +1143,91 @@ xrdsst member list-classes --instance <XROAD-INSTANCE>
...
+
+* ID: Local group id.
+* CODE: Local group code.
+* DESCRIPTION: Local group description.
+* MEMBERS: List of members ids
+
+##### 4.2.10.4 Local groups delete
+
+* Access rights: XROAD_SERVICE_ADMINISTRATOR
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md:1203]: Don't you need both roles XROAD_SYSTEM_ADMINISTRATOR and XROAD_SERVICE_ADMINISTRATOR here?

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 6]
```diff
@@ -1119,6 +1143,91 @@ xrdsst member list-classes --instance <XROAD-INSTANCE>
...
+```
+xrdsst local-group delete --ss <SECURITY_SERVER_NAME> --local-group <LOCAL_GROUP_ID>
+```
+* <SECURITY_SERVER_NAME> security server name, e.g. ss1
+* <LOCAL_GROUP_ID> Local group id (can be check with the command [4.2.10.3 Local groups list](#42103-local-groups-list)), 
+  multiple values can also be given, separated by comma, e.g. 125,127
+
+##### 4.2.10.5 Local groups member delete
+
+* Access rights: XROAD_SERVICE_ADMINISTRATOR
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md:1215]: Is this single role enough for that operation?

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 7]
```diff
@@ -1119,6 +1143,91 @@ xrdsst member list-classes --instance <XROAD-INSTANCE>
...
+
+Delete client local group members can be done with:
+```
+xrdsst local-group delete --ss <SECURITY_SERVER_NAME> --local-group <LOCAL_GROUP_ID> --member <MEMBERS_ID>
+```
+id of the service description, e.g. 123, multiple values can also be given, separated by comma, e.g. 123,456
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md]: This seems to be some kind of copy-paste and does not belong here?

File: docs/xroad_security_server_toolkit_user_guide.md [hunk 8]
```diff
@@ -1119,6 +1143,91 @@ xrdsst member list-classes --instance <XROAD-INSTANCE>
...
+
+* <SECURITY_SERVER_NAME> seccurity server name, e.g. ss1
```
Comment (bertvi) [inline docs/xroad_security_server_toolkit_user_guide.md]: "seccurity" should be "security"

File: xrdsst/controllers/local_group.py [hunk 9]
```diff
@@ -0,0 +1,319 @@
+from cement import ex
```
Comment (bertvi) [inline xrdsst/controllers/local_group.py:1]: This file shows 12 PEP8 warnings (your IDE should probably show these on the right as yellow markers)

File: xrdsst/controllers/client.py [hunk 10]
```diff
@@ -353,6 +353,12 @@ def find_client(self, clients_api, client_conf):
         return found_clients[0]
 
     @staticmethod
+    def find_all_clients(clients_api, show_members=False, internal_search=False):
+        try:
+            return clients_api.find_clients(show_members=show_members, internal_search=internal_search)
+        except ApiException as find_err:
+            BaseController.log_api_error('ClientsApi->find_clients', find_err)
+    @staticmethod
```
Comment (bertvi) [inline xrdsst/controllers/client.py:362]: This class also seems to have a couple of PEP8 warnings, for example this method does not have a blank line before the other method

**Review by bertvi — CHANGES_REQUESTED** [review]

File: docs/xroad_security_server_toolkit_user_guide.md (hunk 3 above)
Comment (bertofl90) [reply docs/xroad_security_server_toolkit_user_guide.md:1183]: it's only prepared for one client each time

File: docs/xroad_security_server_toolkit_user_guide.md (hunk 5 above)
Comment (bertofl90) [reply docs/xroad_security_server_toolkit_user_guide.md:1203]: no only the service administrator

File: docs/xroad_security_server_toolkit_user_guide.md (hunk 6 above)
Comment (bertofl90) [reply docs/xroad_security_server_toolkit_user_guide.md:1215]: yes

**Review by bertvi — APPROVED** [review]

**Pull request merged**